linkerd/service-profiles/src/http/route_request.rs

```diff
@@ -0,0 +1,130 @@
+use super::{Receiver, RequestMatch, Route};
+use crate::Profile;
+use futures::{future::ErrInto, prelude::*, ready};
+use linkerd2_error::Error;
+use linkerd2_stack::{layer, NewService, Proxy};
+use std::{
+    collections::HashMap,
+    marker::PhantomData,
+    task::{Context, Poll},
+};
+use tracing::{debug, trace};
+
+pub fn layer<M, N: Clone, R>(
+    new_route: N,
+) -> impl layer::Layer<M, Service = NewRouteRequest<M, N, R>> {
+    let default = Route::default();
+    layer::mk(move |inner| NewRouteRequest {
+        inner,
+        new_route: new_route.clone(),
+        default: default.clone(),
```

Comment:
nit: i was about to ask why we clone this around instead of just calling `Route::default`, but then i saw this comment in the removed code:
https://github.com/linkerd/linkerd2-proxy/blob/bf0c222aef3a388d502f6cba5895e8efc48d92df/linkerd/service-profiles/src/http/service.rs#L32-L33

might be worth bringing that back, or something similar?